src/gt4py/backend/dawn_backends.py

```diff
@@ -107,19 +138,38 @@ def visit_VarRef(self, node: gt_ir.VarRef, **kwargs):
         return sir_utils.make_var_access_expr(name=node.name, is_external=True)
 
     def visit_FieldRef(self, node: gt_ir.FieldRef, **kwargs):
-        offset = [node.offset[ax] for ax in DOMAIN_AXES]
+        offset = [node.offset[ax] if ax in node.offset else 0 for ax in DOMAIN_AXES]
         return sir_utils.make_field_access_expr(name=node.name, offset=offset)
 
     def visit_UnaryOpExpr(self, node: gt_ir.UnaryOpExpr, **kwargs):
-        op = node.op.python_symbol
+        op = self.visit_Operator(node.op)
         operand = self.visit(node.arg)
         return sir_utils.make_unary_operator(op, operand)
 
     def visit_BinOpExpr(self, node: gt_ir.BinOpExpr, **kwargs):
         left = self.visit(node.lhs)
         right = self.visit(node.rhs)
-        op = node.op.python_symbol
-        return sir_utils.make_binary_operator(left, op, right)
+        if node.op.python_symbol == "**":
+            sir = self.visit_ExpOpExpr(left, right)
+        else:
+            op = self.visit_Operator(node.op)
+            sir = sir_utils.make_binary_operator(left, op, right)
+        return sir
+
+    def visit_ExpOpExpr(self, left, right):
+        exponent = right.value
+        if exponent == "0":
+            return sir_utils.make_literal_access_expr("1", type=SIR.BuiltinType.Integer)
+        if exponent == "1":
+            return sir_utils.make_unary_operator("+", left)
+        if exponent == "2":
+            return sir_utils.make_binary_operator(left, "*", left)
+        elif exponent == "3":
+            return sir_utils.make_binary_operator(
+                left, "*", sir_utils.make_binary_operator(left, "*", left)
+            )
+        else:
```

Comment:
Should the frontend take care of this optimization? I think this kind of analysis and substitutions should be done by the toolchain.

Reply:
Would this be a new class in `gtscript_frontend`, e.g., `PowerExprOptimizer`?

Reply:
My view on this is that this optimization does not belong to the frontend. I think the frontend should create an IR according to the DSL code, without introducing optimizations which can be applied later by the analysis toolchain.

Reply:
Thank you. We should wait for the math functions to be implemented before adding this optimization to the toolchain. I will remove it for now.